utils/forest_utils/src/io/writer_checksum.rs

```diff
@@ -67,6 +67,42 @@ impl<D: Digest, W> AsyncWriterWithChecksum<D, W> {
     }
 }
 
+/// A void writer that does nothing but implements [AsyncWrite] and [Checksum]
+#[derive(Debug, Clone, Default)]
+pub struct VoidAsyncWriterWithChecksum<D> {
```

Comment:
We could make it not so dry. I mean, don't save anything but still generate a correct checksum. No strong opinion, though.

Reply:
Maybe rename it to `VoidAsyncWriterWithVoidChecksum`? For this particular case we don't really need the checksum, and we could implement what u suggest with `VoidAsyncWriterWithChecksum` when we need it. Sounds good?

Reply:
Fixed.